forum/views.py

```diff
@@ -1,3 +1,69 @@
-from django.shortcuts import render
+from django.views import generic
+from django.views.generic import View, TemplateView, DetailView, ListView, CreateView
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login
+from .models import Forum, Post, Comment, PostVote, CommentVote
+from django.shortcuts import render, redirect, HttpResponse, get_object_or_404
+from .forms import RegisterForm, LoginForm
 
-# Create your views here.
+
+# Can change "standard query reference from object_list to whatever as below"
+class HomePageView(ListView):
+    template_name = "forum/list.html"
+    context_object_name= "post_list"
+
+    # Throwing error on my end for comment, but view works
+    #Should be first page
+    def get_queryset(self):
+        # Not sure how to pass below...need to refractor into class createview maybe?  Sending context forward as is creates error
+        vote_count = Post.objects.first().votes.count()
+        upvotes = Post.objects.first().votes.filter(vote=1).count()
+        downvotes = Post.objects.first().votes.filter(vote=-1).count()
+        total = upvotes - downvotes
+        context = {
+            "vc": vote_count,
+            "total":total
+        }
+        print(context)
+        return Post.objects.all()
+
+
+def posttest(request):
```

Comment:
This is not a good name, and this function won't work if you're not authenticated. Wrapping the whole thing in a `try-except` block is considered bad practice because you won't know what kind of error is raised if an unhandled exception occurs.